### PR TITLE
Swap babel for ESBuild and upgrade to Webpack 5

### LINKED ISSUE #204: feat: Upgrade the default site template to Webpack 5
## Summary

The default site generated by `bridgetown new` currently runs on Webpack 4. Webpack 5 has been out for a couple of months now and I think we should support it by default.

## Reference-level explanation

It should be a straightforward upgrade "in theory". The only blocker I encountered was the `ManifestPlugin` doesn't seem to quite work with Webpack 5. Even when using v3 of the plugin which is supposed to support Webpack 5; I ran into a weird issue where it prefixed `auto` to every path in the manifest for some reason. Not sure if this was due to Webpack or the plugin.

I got around this issue in my projects by using the `AssetsPlugin` instead. I discovered it while reading [this Twitter thread](https://twitter.com/garybernhardt/status/1331051655366799360?s=20) from Gary Bernhardt. The format of the JSON output from this plugin is a bit different though so I had to monkey patch some Bridgetown code to get it working.

 You can see my setup here: https://github.com/ayushn21/static-site-template. The monkey patch is in `plugins/monkey_patches`.

## Unresolved Questions

How and when should we proceed with this? Obviously the preferred option would be to get the `ManifestPlugin` working if possible, but should we consider switching to the `AssetsPlugin` at all? 

I remember @andrewmcodes saying he was also looking into this. I thought I'd create this issue so we can discuss how and when to proceed with this!

### PR DESCRIPTION
**Pull request by ayushn21**
This is a 🙋 feature or enhancement.

I thought it'd be a good idea to swap Babel for ESBuild since it's much faster. (I don't have anything more than anecdotal experience that says its better though so happy to revert if required).

I've also reorganized the default webpack configuration by breaking a bunch of stuff out into variables which hopefully makes it cleaner and easier to read!

This needs a bit more testing before merging as well. I've verified that a new site with Sass works but just need to check configurations and a postcss website. I don't foresee any problems though.

- [x] Test new postcss site
- [x] Test webpack related bundled configurations
- [x] Install `esbuild` packages in webpack update command

Closes #204

### CODE REVIEW

**Comment by jaredcwhite** [comment]
Ooooooooo.

(That is all. I don't have anything to add, other than, sounds awesome!)

File: bridgetown-core/lib/bridgetown-core/configurations/tailwindcss.rb [hunk 1]
```diff
@@ -21,9 +21,14 @@
 
 copy_file "#{TEMPLATE_PATH}/postcss.config.js", "postcss.config.js", force: true
 
-prepend_to_file "frontend/styles/index.css",
-                File.read("#{TEMPLATE_PATH}/css_imports.css")
-
 run "bundle exec bridgetown configure purgecss"
 
+if File.exist?("frontend/styles/index.css")
```
Comment (ayushn21) [inline bridgetown-core/lib/bridgetown-core/configurations/tailwindcss.rb:26]: Just guarding against a case where `index.css` doesn't exist and printing out what the user needs to include in their CSS.

Unlikely to happen but could be the case if someone created a site using sass and then converted to postcss. We don't update the CSS/JS setup when enabling postcss on an existing site as it'd be too complicated.

**Comment by ayushn21** [comment]
Hey @jaredcwhite, I think this is good to go. Let me know if you have any feedback! Thanks :)

File: bridgetown-core/lib/bridgetown-core/commands/webpack.rb [hunk 2]
```diff
@@ -41,11 +41,18 @@ def self.destination_root
         config.root_dir
       end
 
+      protected
```
[the remark this answers is not in the document]
Comment (ayushn21) [reply bridgetown-core/lib/bridgetown-core/commands/webpack.rb:44]: Nothing to do with the crashes whatsoever; but an improvement nonetheless I reckon

**Comment by jaredcwhite** [comment]
I like the idea of switching to `esbuild` and away from `babel` but it's potentially a breaking change so I'd like to test it out on a few sites first. (Although I guess nobody's "required" to upgrade their existing Webpack setup.)

Would definitely be cool to get this in the next release.

**Comment by ayushn21** [comment]
Sure, take your time. I don't anticipate it breaking any sites but you never know so best to be safe!

**Comment by ayushn21** [comment]
Upgrading to Webpack 5 turned out to be easier than I thought. Much easier than my last attempt about 6 months ago. Clearly I've learned how to tame the beast! :D

Anyway I've tacked that onto this PR since the changes were pretty small. But that does mean we need to test this quite thoroughly. I'll test it on a few of my websites in the coming days.

**Comment by ayushn21** [comment]
I tried this on 3 of my websites and it worked just fine!

**Comment by jaredcwhite** [comment]
@ayushn21 Sounds great. I'll prioritize getting this into the next release (0.21.1).

**Comment by jaredcwhite** [comment]
@ayushn21 We probably need an upgrade guide or some kind of documentation somewhere. For example I'd like to update the BT website itself but not quite sure what I do. Or is it as easy as just running `webpack update`?

**Comment by ayushn21** [comment]
Yup it's just `bridgetown webpack update`! ... but run `bridgetown webpack setup` first if you haven't got the included default config set up yet.

**Comment by jaredcwhite** [comment]
@ayushn21 Website all ready to go! I also swapped the old `node-sass` out for newer `sass` and all seems well.

**Pull request merged**